sys/include/can/can.h

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright (C) 2016 OTA keys S.A.
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @ingroup     can
+ * @defgroup    can_dll Data Link Layer
+ * @brief       CAN Data Link Layer
+ *
+ * The Data Link Layer is composed of the device, router, pkt and dll files.
+ * It can be used to send and receive raw CAN frames through multiple CAN controllers.
+ *
+ * @{
+ *
+ *
+ * @file
+ * @brief       Definitions high-level CAN interface
+ *
+ * @author      Vincent Dupont <vincent@otakeys.com>
+ * @author      Toon Stegen <toon.stegen@altran.com>
+ */
+
+#ifndef CAN_H
+#define CAN_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <stdint.h>
+
+#if defined(__linux__)
+
+#include <linux/can.h>
+#include <libsocketcan.h>
+
+#else
+
+/**
+ * @brief Max data length for a CAN frame
+ */
+#define CAN_MAX_DLEN (8)
+
+/**
+ * @name CAN_ID flags and masks
+ * @{
+ */
+/* special address description flags for the CAN_ID */
+#define CAN_EFF_FLAG (0x80000000U) /**< EFF/SFF is set in the MSB */
+#define CAN_RTR_FLAG (0x40000000U) /**< remote transmission request */
+#define CAN_ERR_FLAG (0x20000000U) /**< error message frame */
+
+/* valid bits in CAN ID for frame formats */
+#define CAN_SFF_MASK (0x000007FFU) /**< standard frame format (SFF) */
+#define CAN_EFF_MASK (0x1FFFFFFFU) /**< extended frame format (EFF) */
+#define CAN_ERR_MASK (0x1FFFFFFFU) /**< omit EFF, RTR, ERR flags */
+/** @} */
+
+/**
+ * @brief CAN operational and error states
+ */
+ enum can_state {
+     CAN_STATE_ERROR_ACTIVE = 0,     /**< RX/TX error count < 96 */
+     CAN_STATE_ERROR_WARNING,        /**< RX/TX error count < 128 */
+     CAN_STATE_ERROR_PASSIVE,        /**< RX/TX error count < 256 */
+     CAN_STATE_BUS_OFF,              /**< RX/TX error count >= 256 */
+     CAN_STATE_STOPPED,              /**< Device is stopped */
+     CAN_STATE_SLEEPING,             /**< Device is sleeping */
+     CAN_STATE_MAX
+ };
+
+/**
+ * @brief Controller Area Network Identifier structure
+ *
+ * bit 0-28  : CAN identifier (11/29 bit)
```

Comment:
a comment if the 11-bit ID is left or right aligned in those 29 bits would be useful...